### PR TITLE
allow users to specify agp workspace from an env variable.

### PR DESCRIPTION
**Pull request by jdochez**
it makes it easier than always relying on a gradle property.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -15,12 +15,15 @@ include("tests")
...
 // studio-main workspace.
-val workspaceLocationProperty = providers.gradleProperty("com.android.workspace.location")
-if (!workspaceLocationProperty.isPresent) {
-    throw java.lang.IllegalArgumentException(
-        "com.android.workspace.location Gradle property must point to your checked out studio-main workspace")
+val agpWorkspaceLocationProperty = providers.gradleProperty("com.android.workspace.location")
+val agpWorkspaceLocation  = if (agpWorkspaceLocationProperty.isPresent) {
+    agpWorkspaceLocationProperty.get()
+} else {
+    System.getenv("WORKSPACE_LOCATION") ?:
```
Comment (scott-pollom) [inline settings.gradle.kts]: should "WORKSPACE_LOCATION" be more specific? seems like it might collide with an existing env var

Comment (jdochez) [reply settings.gradle.kts]: renamed to AGP_WORKSPACE_LOCATION

**Review by scott-pollom — APPROVED** [review]

File: settings.gradle.kts [hunk 2]
```diff
@@ -19,7 +19,7 @@ val agpWorkspaceLocationProperty = providers.gradleProperty("com.android.workspa
 val agpWorkspaceLocation  = if (agpWorkspaceLocationProperty.isPresent) {
     agpWorkspaceLocationProperty.get()
 } else {
-    System.getenv("WORKSPACE_LOCATION") ?:
+    System.getenv("AGP_WORKSPACE_LOCATION") ?:
```
Comment (scott-pollom) [inline settings.gradle.kts:22]: Should the other usages of WORKSPACE_LOCATION be updated too?

https://github.com/search?q=repo%3Aandroid%2Fgradle-declarative%20WORKSPACE_LOCATION&type=code

Comment (jdochez) [reply settings.gradle.kts:22]: Done !

**Review by scott-pollom — APPROVED** [review]

**Pull request merged**